### DISCUSSION TITLE
Can behaveX be used as a static analysis metrics tool?

### DISCUSSION THREAD

**Discussion by dgutson** [hrcorval/behavex#107] [Q&A] [answered]
Can I use behaveX to obtain a metrics report without executing the tests, that is, as a static analysis?

For example, metrics I am interested of: # of automated vs # of manual tests per scenario/tag (given a set of tags).

@hrcorval ?

**Comment by hrcorval** [comment, marked as answer]
Hi @dgutson !!! As a first approach to what you need, have you considered performing a dry run? You can launch the framework using the same arguments as in a normal execution, by adding and filtering tags to determine the scenarios to run, but adding also the "--dry-run" argument. This will generate the test execution report without running the tests, and in the report wou can see metrics like the automation rate.
Of course we are plenty o areas for improvements, but this might work as an initial step.
Please, let me know if this works for you.

Any additional suggestions are welcome.

Thanks
